ringpop_test.go

```diff
@@ -147,6 +153,96 @@ func (s *RingpopTestSuite) TestRingpopNotReady() {
 	s.False(s.ringpop.Ready())
 }
 
+// TestStateCreated tests that Ringpop is in a created state just after
+// instantiating.
+func (s *RingpopTestSuite) TestStateCreated() {
+	s.Equal(created, s.ringpop.getState())
+}
+
+// TestStateInitialized tests that Ringpop is in an initialized state after
+// a failed bootstrap attempt.
+func (s *RingpopTestSuite) TestStateInitialized() {
+
+	// Create channel and start listening so we can actually attempt to
+	// bootstrap
+	ch, _ := tchannel.NewChannel("test2", nil)
+	ch.ListenAndServe("127.0.0.1:0")
```

Comment:
`defer` runs at the end of the method, which is not what we want.

We want it to run at the end of the test, which is what TearDownTest does. I haven't confirmed, but I'd expect this is is always run, even if the test panics.